### DISCUSSION TITLE
How to set ECI designator

### DISCUSSION THREAD

**Discussion by shimbala** [chillerlan/php-qrcode#155] [Q&A] [answered]
Hi, 

I need to set ECI value to 4. Is there a way to do that?

Thanks

**Comment by codemasher** [comment, marked as answer]
Hey, I haven't tested the ECI support yet, but with the current `dev-main` you can do it like this (here's [the list of currently supported ECI charsets](https://github.com/chillerlan/php-qrcode/blob/main/src/Common/ECICharset.php#L21-L51)):

```php
$qrcode->addEciDesignator(ECICharset::ISO_IEC_8859_2);
```

You can also add an ECI segment with encoding and data:

```php
$qrcode->addEciSegment(ECICharset::ISO_IEC_8859_2, 'data'); // data will be encoded internally
```

However, the ECI decoding part is not yet properly implemented in the included QR Code reader, also no guarantee yet that the ECI designator works as intended.

**Reply by shimbala** [reply]
Thank you for answer.

**Reply by codemasher** [reply]
Happy to help!
